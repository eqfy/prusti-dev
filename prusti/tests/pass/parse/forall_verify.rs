// compile-flags: -Zprint-desugared-specs -Zprint-typeckd-specs -Zhide-uuids
// normalize-stdout-test: "[a-z0-9]{32}" -> "$(NUM_UUID)"
// normalize-stdout-test: "[a-z0-9]{8}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{12}" -> "$(UUID)"

#![feature(register_tool)]
#![register_tool(prusti)]

use prusti_contracts::*;

#[pure]
fn identity(x: u32) -> u32 { x }

#[ensures(forall(|x: u32| true))]
fn test1() {}

#[ensures(forall(|x: u32| identity(x) == x))]
fn test2() {}

#[ensures(forall(|x: u32| identity(x) == x + 1))]
fn test2() {}

fn main() {}
