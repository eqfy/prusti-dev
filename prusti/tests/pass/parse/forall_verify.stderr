error[E0428]: the name `test2` is defined multiple times
  --> $DIR/forall_verify.rs:20:1
   |
17 | fn test2() {}
   | ---------- previous definition of the value `test2` here
...
20 | fn test2() {}
   | ^^^^^^^^^^ `test2` redefined here
   |
   = note: `test2` must be defined only once in the value namespace of this module

error: aborting due to previous error

For more information about this error, try `rustc --explain E0428`.
